src/Components/Productpage/ProductListing/ProductList.jsx

```diff
@@ -38,22 +38,25 @@ const ProductList = ({ products }) => {
 
   useEffect(() => {
     setTimeout(() => {
-      setShowAlert(false);
+      setShowAlert({
+        showAlert: false,
+        alertMessage: null,
+        type: null,
+      });
     }, 2000);
```

Comment:
I would suggest to clearTheSetTimeOut As it running whenever the showAlert will change.

Because when the component will unmount you can clear out the previous effects in your case setTimout.